libraries/Matter/examples/MatterComposedLights/MatterComposedLights.ino

```diff
@@ -0,0 +1,103 @@
+MatterOnOffLight OnOffLight3;
+
+// Matter Protocol Endpoint Callback for each Light Accessory
+bool setLightOnOff1(bool state) {
+  Serial.printf("CB-Light1 changed state to: %s\r\n", state ? "ON" : "OFF");
+  return true;
+}
+
+bool setLightOnOff2(bool state) {
+  Serial.printf("CB-Light2 changed state to: %s\r\n", state ? "ON" : "OFF");
+  return true;
+}
+
+bool setLightOnOff3(bool state) {
+  Serial.printf("CB-Light3 changed state to: %s\r\n", state ? "ON" : "OFF");
+  return true;
+}
+
+// WiFi is manually set and started
+
+const char *ssid = "your-ssid";          // Change this to your WiFi SSID
+const char *password = "your-password";  // Change this to your WiFi password
+
+void setup() {
+  Serial.begin(115200);
+  while (!Serial) {
+    delay(100);
+  }
+
+  // We start by connecting to a WiFi network
+  Serial.print("Connecting to ");
+  Serial.println(ssid);
+  // enable IPv6
+  WiFi.enableIPv6(true);
+  // Manually connect to WiFi
+  WiFi.begin(ssid, password);
+  // Wait for connection
+  while (WiFi.status() != WL_CONNECTED) {
+    delay(500);
+    Serial.print(".");
+  }
+  Serial.println("\r\nWiFi connected");
+  Serial.println("IP address: ");
+  Serial.println(WiFi.localIP());
+  delay(500);
+
+  // Initialize all 3 Matter EndPoints
+  OnOffLight1.begin();
+  OnOffLight2.begin();
+  OnOffLight3.begin();
+  OnOffLight1.onChangeOnOff(setLightOnOff1);
+  OnOffLight2.onChangeOnOff(setLightOnOff2);
+  OnOffLight3.onChangeOnOff(setLightOnOff3);
+
+  // Matter begining - Last step, after all EndPoints are initialized
+  Matter.begin();
+}
+
+void loop() {
+  // Check Matter Light Commissioning state
+  if (!Matter.isDeviceCommissioned()) {
+    Serial.println("");
+    Serial.println("Matter Node is not commissioned yet.");
+    Serial.println("Initiate the device discovery in your Matter environment.");
+    Serial.println("Commission it to your Matter hub with the manual pairing code or QR code");
+    Serial.printf("Manual pairing code: %s\r\n", Matter.getManualPairingCode().c_str());
+    Serial.printf("QR code URL: %s\r\n", Matter.getOnboardingQRCodeUrl().c_str());
+    // waits for Matter Light Commissioning.
+    uint32_t timeCount = 0;
+    while (!Matter.isDeviceCommissioned()) {
+      delay(100);
+      if ((timeCount++ % 50) == 0) { // 50*100ms = 5 sec
+        Serial.println("Matter Node not commissioned yet. Waiting for commissioning.");
+      }
+    }
+    Serial.println("Matter Node is commissioned and connected to Wi-Fi. Ready for use.");
+  }
+
+  //displays the Light state every 3 seconds
+  Serial.println("======================");
+  Serial.printf("Matter Light #1 is %s\r\n", OnOffLight1.getOnOff() ? "ON" : "OFF");
+  Serial.printf("Matter Light #2 is %s\r\n", OnOffLight2.getOnOff() ? "ON" : "OFF");
+  Serial.printf("Matter Light #3 is %s\r\n", OnOffLight3.getOnOff() ? "ON" : "OFF");
+  delay(3000);
+}
+
+#else
+void setup() {}
+void loop() {}
+#endif /* CONFIG_ESP_MATTER_ENABLE_DATA_MODEL */
```

Comment:
I don't think this is necessary as the macro is already required in the JSON to compile the sketch

Reply:
If I remove it, it fails with ESP32S3 when CI compiles the examples.

Reply:
But does S3 have support for Matter ? If yes, then something is wrong somewhere.

Reply:
Yes, this is Matter over WiFi. It is suppose to work with ESP32, S2, S3, C3 and C6.
CI seems to pick some `sdkconfig.h` that has no Matter configuration, different from the current `release/v3.1.x` file that I get when I run `get.py` in my computer.

Reply:
The commit that are right after this comment https://github.com/espressif/arduino-esp32/pull/10467#issuecomment-2415411880 are about trying to overcome CI errors.

I have tried many different possibilities. None of them have worked. 
The only way to get it to pass CI was using the current code.

Reply:
If the sdkconfig didn't include `CONFIG_ESP_MATTER_ENABLE_DATA_MODEL` it wouldn't even try to compile the sketch for the chip. What was the issue exactly ? I will fix any CI related issues.

Reply:
`#define CONFIG_ESP_MATTER_ENABLE_DATA_MODEL 1` only exists for QIO. The CI uses DIO by default. Overwrite the FQBNs and it should work properly.

Reply:
Maybe we can move to using QIO by default as, with the new JSON options, it's easy to build in DIO if needed. I'll open a PR for that

Reply:
Why DIO/QIO (flash mode) has to do with which `sdkconfig` is used? Is it something specific from CI? Please tell more about it.

Reply:
Please check these 2 commit:
5960abede4c31e3ba2da79adf27dadafa2b468ab
9e74283a90cdc528e2e82efee091737c1be75bc7

Is it what you are talking about?

Reply:
Overwriting CI FBQN for S3 with:

``` json
{
    "fqbn": {
      "esp32s3": [
        "espressif:esp32:esp32s3:PartitionScheme=huge_app,FlashMode=qio"
      ]
    },
    "requires": [
      "CONFIG_SOC_WIFI_SUPPORTED=y",
      "CONFIG_ESP_MATTER_ENABLE_DATA_MODEL=y"
    ]
}
```

Reply:
> Why DIO/QIO (flash mode) has to do with which `sdkconfig` is used? Is it something specific from CI? Please tell more about it.

I also don't know, but the libs are only being built for QIO as you can see:

<img width="647" alt="image" src="https://github.com/user-attachments/assets/aae4bcf2-2ddf-4901-a41e-1c237110a19d">

Reply:
It was a problem with Lib Builder. It is fixed now and the new lib files shall be available by tomorrow.
I'll remove all unnecessary guards and ci.json content.